### DISCUSSION TITLE
Running nvim-dap as root

### DISCUSSION THREAD

**Discussion by TimKirathe** [mfussenegger/nvim-dap#1300] [Q&A]
Hello! I'm currently working on a c++ project that requires my debugger to be run as root in order to work. I'm using vscode-cpptools, with the nvim-dap-ui plugin as well. Is there is a way to make nvim-dap run my debugger as root/with root privileges?

**Comment by shmerl** [comment]
I don't think DAP supports that. You can run gdbserver yourself with any privileges you want, and then connect to it using gdbserver path in DAP adapters.

And why do you need privileges in general? It doesn't make sense unless you need gdb itself to attach to something. And even then, you can relax yama temporarily.

See: https://www.kernel.org/doc/html/latest/admin-guide/LSM/Yama.html

**Reply by TimKirathe** [reply]
Thanks for your reply. 

I need root because im using a low-level library that requires root access in order to use it's functions. Before I used nvim-dap, I would run the executable with gdb as root on a different terminal window and then do all debugging from there. But because I did debugging very often it became a hassle to keep writing break points in different parts of my file again and again after every run. Problem is that when I run nvim-dap from within neovim, it runs gdb as a user and not root so I can't proceed with anything. I could however, as you mentioned, run gdbserver as root and then connect to it within nvim-dap. Thanks for the info
